src/parsers/Parse_box_shadow.re

```diff
@@ -0,0 +1,123 @@
+[@bs.module "./normalizeColor"]
+external normalizeColor: string => Js.Nullable.t(float) = "normalizeColor";
+
+let outerWhitespaceRe = [%re "/(?!\(.*)\s(?![^(]*?\))/g"];
+let outerCommaRe = [%re "/(?!\(.*),(?![^(]*?\))/g"];
+
+let testBoxShadow = val_ => {
+  switch (val_) {
+  | "none" => true
+  | _ =>
+    let properties =
+      Js.String.splitByRe(outerWhitespaceRe, val_)
+      |> Js.Array.filter(str => {
+           switch (str) {
+           | Some(s) =>
+             s !== "inset" && normalizeColor(s) === Js.Nullable.null
+           | None => false
+           }
+         });
+
+    switch (Array.length(properties)) {
+    | n when n >= 2 && n <= 4 =>
+      properties
+      |> Js.Array.every(p =>
+           p->Belt.Option.getWithDefault("") |> Parse_unit.testUnit
+         )
+    | _ => false
+    };
+  };
+};
+
+let testBoxShadows = val_ => {
+  let boxShadows = Js.String.splitByRe(outerCommaRe, val_);
```

Comment:
Seems like you’ll need this for transforms then?

Reply:
I have a specific regex for transforms, but it's quite similar! It's down in `Parse_transform`. The goal for this regex is to split on commas not inside parenthesis. So it handles when a user supplies multiple box-shadows that use `rgba` or `hsla` colors (colors don't get split, but the comma separating the individual box-shadows gets picked up).